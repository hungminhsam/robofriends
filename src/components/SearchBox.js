import React from "react";
// import { connect } from "react-redux";
// import { setSearchField } from "../redux/actions";

const SearchBox = (props) => {
  return (
    <div className="pa2">
      <input
        className="pa3 ba b-green bg-lightest-blue"
        type="search"
        placeholder="Search Robots"
        onChange={(e) => props.onChange(e.target.value.trim())}
      />
    </div>
  );
};

// export default connect(null, { setSearchField })(SearchBox);
export default SearchBox;
