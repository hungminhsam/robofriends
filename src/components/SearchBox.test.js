import { shallow } from "enzyme";
import React from "react";
import { SearchBox } from "./SearchBox";

describe("<SearchBox />", () => {
  it("expect to render SearchBox", () => {
    const mockProps = {
      searchText: "",
      changeSearchText: jest.fn(),
    };
    expect(shallow(<SearchBox {...mockProps} />).debug()).toMatchSnapshot();
  });

  it("expect to render SearchBox with correct SearchText passed as props", () => {
    const mockProps = {
      searchText: "search text",
      changeSearchText: jest.fn(),
    };
    expect(
      shallow(<SearchBox {...mockProps} />)
        .find("input")
        .prop("value")
    ).toEqual("search text");
  });
});
