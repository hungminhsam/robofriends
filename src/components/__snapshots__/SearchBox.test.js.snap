// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBox /> expect to render SearchBox 1`] = `
"<div className=\\"pa2\\">
  <input aria-label=\\"Search Robots\\" className=\\"pa3 ba b-green bg-lightest-blue\\" type=\\"search\\" placeholder=\\"Search Robots\\" value=\\"\\" onChange={[Function: onChange]} />
</div>"
`;
